scripts/scr_unit_size/scr_unit_size.gml

```diff
@@ -1,23 +1,30 @@
-
-function scr_unit_size(armour, role, other_factors) {
+
+// lets face it this should all be solved in a method in the marines struct....
+function scr_unit_size(armour, role, other_factors, mobility) {
 
     // armour: armour
     // role: role
     // other_factors: +?
+	//mobility item
     var sz = 1;
     if (role!=""){

```

Comment:
you can simplify this, by inverting this if and making it a guard clause:

```
if(role == "") then return 0; 
...
```
it will make reading easier and simplify the flow a bit